deutsches-archaologisches-institut.csl

```diff
@@ -251,37 +246,28 @@
       <text variable="locator"/>
     </group>
   </macro>
-  <macro name="container-title-short">
-    <group>
-      <choose>
-        <if type="article-journal">
-          <choose>
-            <if variable="container-title-short" match="none">
-              <text variable="container-title"/>
-            </if>
-          </choose>
-          <text variable="container-title-short"/>
-        </if>
-      </choose>
-    </group>
-  </macro>
   <macro name="date">
     <date variable="issued">
       <date-part name="year"/>
     </date>
   </macro>
   <macro name="place-date">
     <choose>
-      <if type="book thesis" match="any">
-        <text variable="publisher-place" prefix=" (" suffix=" "/>
+      <if type="book thesis chapter" match="any">
+        <choose>
+          <if match="none" is-numeric="edition">
+            <text value=" "/>
+          </if>
+        </choose>
```

Comment:
This conditional prefix here is rather awkward.

In general, it's much better to have macros output a block of content without any separating prefixes or suffixes surrounding the block. E.g. "(Berlin, 2009)" instead of " (Berlin, 2009)". You can then control the delimiters when calling the macros:

```
        <choose>
          <if is-numeric="edition">
            <group delimiter=", ">
              <text macro="series"/>
              <text macro="edition"/>
            </group>
            <text macro="place-date"/>
          </if>
          <else>
            <group delimiter=" ">
              <text macro="series"/>
              <text macro="place-date"/>
             </group>
          </else>
        </choose>
```

That makes it much easier to understand what's going on.